exercise-book/src/cpp-interop.md

```diff
@@ -0,0 +1,349 @@
+# Cpp Interop Exercise
+
+## After completing this exercise you will be able to
+
+- use `autocxx` to deal with C++ data types from Rust in a header-only library
+- write your own thin wrappers for C++ functions with default parameters
+- be able to call template C++ functions from Rust with `autocxx`
+
+
+## Prerequisites
+
+- FFI
+- Some knowledge of C/C++
+
+This exercise does not require writing vast amounts of Rust code, but does require understanding how different parts of the build system in Rust and C++ interact to get the project going.
+
+## Tasks
+
+1. Use `autocxx` to develop bindings to a `rapidcsv.h` and print out the dates in `example.csv`, from within Rust as well as the `RowCount`.
+
+You should get:
+
+```console
+6
+2017-02-24
+2017-02-23
+2017-02-22
+2017-02-21
+2017-02-17
+```
+
+Because there are 5 rows of data and a header row.
+
+2. Use `autocxx` to read `weather.csv` and report the average temperature in June, which should be approximately
+
+```console
+1.20
+```
+
+A full solution is available at `rust-exercises/exercises-solutions/cpp-interop`.
```

Comment:
Maybe we should expand the tasks section? Give a list of steps to follow within each point to give people a plan to follow?

Reply:
I've expanded both sections now. Care to review for phrasing please?